libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -253,6 +291,67 @@ export class OrchestratorAdaptiveRecognizer
         return recognizerResult;
     }
 
+    private getTopTwoIntents(
+        result: RecognizerResult
+    ): { name: string; score: number }[] {
+        if (!result || !result.intents) {
+            throw new Error('result is empty');
+        }
+        const intents = Object.entries(result.intents)
+                            .map((intent) => {return {name: intent[0], score: +intent[1].score} })
+                            .sort((a,b) => (b.score - a.score))
+        intents.length = 2;
+
+        return intents;
+    }
+
+    /**
+     * Uses the RecognizerResult to create a list of properties to be included when tracking the result in telemetry.
+     * 
+     * @param {RecognizerResult} recognizerResult Recognizer Result.
+     * @param {Record<string, string>} telemetryProperties A list of properties to append or override the properties created using the RecognizerResult.
+     * @param {DialogContext} dialogContext Dialog Context.
+     * @returns {Record<string, string>} A collection of properties that can be included when calling the TrackEvent method on the TelemetryClient.
+     */
+    protected fillRecognizerResultTelemetryProperties(
+        recognizerResult: RecognizerResult,
+        telemetryProperties: Record<string, string>,
+        dialogContext?: DialogContext
+    ): Record<string, string> {
+        const topTwo = this.getTopTwoIntents(recognizerResult);
```

Comment:
`const [topOne, topTwo] = this.getTopTwoIntents(...) ?? [];` extracts those array elements directly, then you can just do `topOne.name, topOne.score`, etc.